homeassistant/components/zigate.py

```diff
@@ -0,0 +1,278 @@
+"""
+ZiGate component.
+
+For more details about this platform, please refer to the documentation
+https://home-assistant.io/components/ZiGate/
+"""
+import logging
+import voluptuous as vol
+import os
+
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.entity_component import EntityComponent
+from homeassistant.helpers.discovery import load_platform
+from homeassistant.const import (ATTR_BATTERY_LEVEL, CONF_PORT,
+                                 EVENT_HOMEASSISTANT_START,
+                                 EVENT_HOMEASSISTANT_STOP)
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+REQUIREMENTS = ['zigate==0.17.3']
+DEPENDENCIES = ['persistent_notification']
+
+DOMAIN = 'zigate'
+DATA_ZIGATE_DEVICES = 'zigate_devices'
+DATA_ZIGATE_ATTRS = 'zigate_attributes'
+ADDR = 'addr'
+
+CONFIG_SCHEMA = vol.Schema({
+    vol.Optional(CONF_PORT): cv.string,
+}, extra=vol.ALLOW_EXTRA)
+
+
+REFRESH_DEVICE_SCHEMA = vol.Schema({
+    vol.Optional(ADDR): cv.string,
+})
+
+RAW_COMMAND_SCHEMA = vol.Schema({
+    vol.Required('cmd'): cv.positive_int,
+    vol.Optional('data'): cv.string,
+})
+
+IDENTIFY_SCHEMA = vol.Schema({
+    vol.Required(ADDR): cv.string,
+})
+
+
+def setup(hass, config):
+    """Setup zigate platform."""
+    from homeassistant.components import persistent_notification
+    import zigate
+
+    port = config.get(CONF_PORT)
+    persistent_file = os.path.join(hass.config.config_dir,
+                                   'zigate.json')
+    _LOGGER.debug('Persistent file {}'.format(persistent_file))
+
+    z = zigate.ZiGate(port,
+                      persistent_file,
+                      auto_start=False)
+
+    hass.data[DOMAIN] = z
+    hass.data[DATA_ZIGATE_DEVICES] = {}
+    hass.data[DATA_ZIGATE_ATTRS] = {}
+
+    component = EntityComponent(_LOGGER, DOMAIN, hass)
+
+    def device_added(**kwargs):
+        device = kwargs['device']
+        _LOGGER.debug('Add device {}'.format(device))
+        if device.addr not in hass.data[DATA_ZIGATE_DEVICES]:
+            entity = ZiGateDeviceEntity(device)
```

Comment:
This type of general device entity will probably be replaced by the device concept in the future:
https://github.com/home-assistant/architecture/issues/36